sql-gremlin/src/main/java/org/twilmes/sql/gremlin/adapter/converter/ast/nodes/select/GremlinSqlSelectMulti.java

```diff
@@ -306,17 +317,152 @@ protected void applyOrderBy(final GraphTraversal<?, ?> graphTraversal, final Str
+    protected void applyWhere(final GraphTraversal<?, ?> graphTraversal,
+                              final String inVRename, final String outVRename) throws SQLException {
+        SqlNode sqlNode = sqlSelect.getWhere();
+        if (sqlNode == null) {
             return;
         }
-        throw SqlGremlinError.createNotSupported(SqlGremlinError.JOIN_WHERE_UNSUPPORTED);
+        applySqlFilter(sqlNode, graphTraversal, inVRename, outVRename);
+    }
+
+    private void applySqlFilter(final SqlNode sqlNode, final GraphTraversal<?, ?> graphTraversal,
+                       final String inVRename, final String outVRename) throws SQLException {
+        if (sqlNode instanceof SqlBasicCall) {
+            SqlBasicCall sqlBasicCall = (SqlBasicCall) sqlNode;
+            if (sqlBasicCall.getOperator() instanceof SqlPrefixOperator) {
+                if (sqlBasicCall.getOperator().kind.equals(SqlKind.NOT)) {
+                    if (sqlBasicCall.getOperandList().size() == 1) {
+                        // if operator == NOT => recursively calling applySqlFilter() and then apply NOT
+                        final GraphTraversal<?, ?> subGraphTraversal = __.__();
+                        applySqlFilter(sqlBasicCall.getOperandList().get(0), subGraphTraversal, inVRename, outVRename);
+                        graphTraversal.not(subGraphTraversal);
+                        return;
+                    }
+                    throw SqlGremlinError.createNotSupported(SqlGremlinError.WHERE_NOT_ONLY_BOOLEAN);
+                }
+                throw SqlGremlinError.createNotSupported(SqlGremlinError.WHERE_UNSUPPORTED_PREFIX);
+            } else if (sqlBasicCall.getOperandList().size() == 2) {
+                if (sqlBasicCall.getOperator().kind.equals(SqlKind.AND) ||
+                        sqlBasicCall.getOperator().kind.equals(SqlKind.OR)) {
+                    // if operator == AND or OR => recursively calling applySqlFilter() and then apply AND or OR
+                    GraphTraversal<?, ?>[] list = new GraphTraversal[2];
+                    for (int i = 0; i < 2; i++) {
+                        SqlNode node = sqlBasicCall.getOperandList().get(i);
+                        final GraphTraversal<?, ?> subGraphTraversal = __.__();
+                        applySqlFilter(node, subGraphTraversal, inVRename, outVRename);
+                        list[i] = subGraphTraversal;
+                    }
+                    if (sqlBasicCall.getOperator().kind.equals(SqlKind.AND)) {
+                        graphTraversal.and(list);
+                    } else {
+                        graphTraversal.or(list);
+                    }
+                    return;
+                }
+                GremlinSqlNode op1 = createNode(sqlBasicCall.getOperandList().get(0));
+                final GremlinSqlLiteral gremlinSqlLiteral = GremlinSqlFactory
+                        .createNodeCheckType(sqlBasicCall.getOperandList().get(1), GremlinSqlLiteral.class);
```

Comment:
Since we are only supporting literal comparisons in WHERE and HAVING clause, I'd suggest throwing an unsupported operation error to user instead of this Type Mismatch error to be more clear. 
```suggestion
                final GremlinSqlLiteral gremlinSqlLiteral;
                try {
                    gremlinSqlLiteral = GremlinSqlFactory
                        .createNodeCheckType(sqlBasicCall.getOperandList().get(1), GremlinSqlLiteral.class);
                } catch (SQLException e) {
                    throw SqlGremlinError.createNotSupported(SqlGremlinError.UNSUPPORTED_BASIC_LITERALS);
                }
```
Because we don't have an error defined for this case with both WHERE and HAVING, in my suggestion I'm using an SQL error enum `UNSUPPORTED_BASIC_LITERALS`, which would need to be added to the `SQLGremlinError.java` file, and the definition for it `UNSUPPORTED_BASIC_LITERALS=Unsupported: Unsupported clause - Only basic literal comparisons are supported inside WHERE and HAVING at this time.` will need to be added to the `error-messages_en.properties` file.